sql/expression/function/json_array.go

```diff
@@ -0,0 +1,143 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package function
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/dolthub/go-mysql-server/sql"
+)
+
+// JSON_ARRAY([val[, val] ...])
+//
+// JSONArray Evaluates a (possibly empty) list of values and returns a JSON array containing those values.
+//
+// https://dev.mysql.com/doc/refman/8.0/en/json-creation-functions.html#function_json-array
+
+type JSONArray struct {
+	Docs []sql.Expression
+}
+
+var _ sql.FunctionExpression = (*JSONArray)(nil)
+
+// NewJSONArray creates a new JSONArray function.
+func NewJSONArray(args ...sql.Expression) (sql.Expression, error) {
+	return &JSONArray{Docs: args}, nil
+}
+
+// FunctionName implements sql.FunctionExpression
+func (j JSONArray) FunctionName() string {
+	return "json_array"
+}
+
+// Description implements sql.FunctionExpression
+func (j JSONArray) Description() string {
+	return "creates JSON array."
+}
+
+// IsUnsupported implements sql.UnsupportedFunctionStub
+func (j JSONArray) IsUnsupported() bool {
+	return false
+}
+
+// Resolved implements the Expression interface.
+func (j *JSONArray) Resolved() bool {
+	for _, d := range j.Docs {
+		if !d.Resolved() {
+			return false
+		}
+	}
+	return true
+}
+
+// String implements the Expression interface.
+func (j *JSONArray) String() string {
+	children := j.Children()
+	var parts = make([]string, len(children))
+
+	for i, c := range children {
+		parts[i] = c.String()
+	}
+
+	return fmt.Sprintf("JSON_ARRAY(%s)", strings.Join(parts, ", "))
+}
+
+// Type implements the Expression interface.
+func (j *JSONArray) Type() sql.Type {
+	return sql.JSON
+}
+
+// IsNullable implements the Expression interface.
+func (j *JSONArray) IsNullable() bool {
+	for _, d := range j.Docs {
+		if d.IsNullable() {
+			return true
+		}
+	}
+	return false
+}
+
+// Eval implements the Expression interface.
+func (j *JSONArray) Eval(ctx *sql.Context, row sql.Row) (interface{}, error) {
+	if len(j.Docs) == 0 {
+		return sql.JSONDocument{Val: make([]interface{}, 0)}, nil
+	}
+
+	var resultArray = make([]interface{}, len(j.Docs))
+
+	for i, doc := range j.Docs {
+		jsonDoc, err := doc.Eval(ctx, row)
+		if err != nil {
+			return nil, err
+		}
+
+		if jsonStr, ok := jsonDoc.(string); ok {
+			resultArray[i] = jsonStr
+		} else {
+			jsonDoc, err = j.Type().Convert(jsonDoc)
+			if err != nil {
+				return nil, err
+			}
+
+			js := jsonDoc.(sql.JSONDocument).Val
+
+			switch js.(type) {
+			case time.Time:
+				t := js.(time.Time)
```

Comment:
Curious about this. Does sql.JSONDocument not convert time to its string representation. Not sure why this is necessary.